### PR TITLE
[Android] Vibrate device when KMW calls "beep"

### LINKED ISSUE #266: [Android] CODE_BEEP should send a vibration or sound
<a href="https://github.com/mcdurdin"></img></a> **Issue by [mcdurdin](https://github.com/mcdurdin)**
_Wednesday Jul 1, 2015 at 22:40 GMT_
_Originally opened as KD-25_

----
Similar to #257 

Since KMW engine handles all codes, it should send a command to vibrate or sound on `CODE_BEEP`

### PR DESCRIPTION
**Pull request by darcywong00**
Fixes #266 

Currently in the mobile viewport of KMW, the screen will flash for `beep`. This adds a 500ms vibrate to the physical device.

Other notes:
* removed a spurious `AndroidManifest.xml file`
* added a simple KMW test page for `beep`.

### CODE REVIEW

File: web/source/kmwcallback.ts [hunk 1]
```diff
@@ -693,6 +693,10 @@ namespace com.keyman {
         this._BeepTimeout = 1;
         window.setTimeout(this.beepReset.bind(this), 50);
       }
+
+      if ('beepKeyboard' in this.keymanweb) {
```
Comment (darcywong00) [inline web/source/kmwcallback.ts]: I'm quite open to naming this something else...

File: android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java [hunk 2]
```diff
@@ -1613,6 +1620,22 @@ public int getKeyboardWidth() {
       return kbWidth;
     }
 
+    // This annotation is required in Jelly Bean and later:
+    @JavascriptInterface
+    public void beepKeyboard() {
+      int duration = 500; // milliseconds
```
Comment (mcdurdin) [inline android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java]: This is definitely too long. When I tested it was quite dramatic :) I tried with 100ms and that felt more appropriate. It needs to be a nudge, not a big serious flag. Can you experiment with durations <100ms to see how they feel as well? What do you think?

File: web/testing/issue266/issue266.js [hunk 3]
```diff
@@ -0,0 +1,1361 @@
+
```
[the remark this answers is not in the document]
Comment (darcywong00) [reply web/testing/issue266/issue266.js]: The other test JS keyboards are in the repo. but I'll remove the kmx

File: android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java (hunk 2 above)
Comment (darcywong00) [reply android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java]: I tried 50ms and 100ms and can't really tell the difference. Will stick with 100ms

**Pull request merged**